### PR TITLE
S22-5pm-1  Add API endpoint for getting course information with detailed section info, in a useable for

### LINKED ISSUE #19: Add API endpoint for getting course information with detailed section info, in a useable form.
- [x] Update`UCSBCurriculumService.java`
- [x] Create `UCSBSectionsController.java` file to return proper JSON on request
- [x] Implement a `Get`  function for controller that returns a JSON file
- [x] Full Test coverage
- [x] 100% Mutation killed

### PR DESCRIPTION
**Pull request by alexward120**
In this pull request, I added an API endpoint to get section information based on quarter, courseLevel, and courseCode. I also added various documents in order for the formatting to work which can be viewed in the folder called documents. All mutation and code coverage.
 
image
Above you can see the controller for this endpoint.
image
This endpoint returns each section in a list of json where each element is a different section
image
image

### CODE REVIEW

**Review by nthnga13 — DISMISSED** [review]
LGTM!

File: pom.xml [hunk 1]
```diff
@@ -170,6 +170,7 @@
                         <param>edu.ucsb.cs156.courses.config.SecurityConfig</param>
                         <param>edu.ucsb.cs156.courses.config.SecurityConfig.MyCsrfRequestMatcher</param>
                         <param>edu.ucsb.cs156.courses.config.SpringFoxConfig</param>
+                        <param>edu.ucsb.cs156.courses.documents.CoursePage</param>
```
Comment (pconrad) [inline pom.xml]: Why are we excluding CoursePage from mutation testing?

This doesn't sound right to me.

I think you need to remove this and do proper testing, unless you can make a solid case that it's not appropriate.

Show me what the mutation test report says when this is removed, and we can discuss it.

And a MINIMUM, please make an issue in your repo that asks the programmer to remove 
```
<param>edu.ucsb.cs156.courses.documents.CoursePage</param>
``` 

and implement appropriate tests.

File: src/main/java/edu/ucsb/cs156/courses/documents/CoursePage.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    /**
+     * Create a CoursePage object from json representation
+     * 
+     * @param json String of json returned by API endpoint {@code /classes/search}
+     * @return a new CoursePage object
+     * @see <a href=
+     *      "https://developer.ucsb.edu/content/academic-curriculums">https://developer.ucsb.edu/content/academic-curriculums</a>
+     */
+    public static CoursePage fromJSON(String json) {
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/documents/CoursePage.java:32]: You have tests for these methods, so why are you excluding the class from test coverage?

**Review by pconrad — CHANGES_REQUESTED** [review]
Other than the exclusion from mutation testing, this LGTM.

But please address that.    Show me what problems you are having with mutation testing and maybe we can come up with a better solution.

File: pom.xml (hunk 1 above)
Comment (alexward120) [reply pom.xml]: When i run it it shows this, it seems to be just line coverage.
image
image

**Review by pconrad — APPROVED** [review]
LGTM...let's try it on QA

**Pull request merged**